### PR TITLE
Fix genPutArgStkFieldList for SIMD12

### LINKED ISSUE #79118: AccessViolationException using System.Numerics.Vector3
### Description

I've noticed a memory corruption bug when using `Vector3` from `System.Numerics`. It seems to be related to passing `Vector3` to functions. It only happens in Release builds.

### Reproduction Steps

The following code will trigger the memory corruption. It is not completely deterministic, but always happens after a second or so. Note that the bug only appears in Release build for me.

Sorry for the slightly lengthy code, I couldn't shorten it further.

```csharp
using System.Numerics;
using System.Runtime.CompilerServices;

readonly struct Ray
{
    public readonly Vector3 Origin;
    public readonly Vector3 Direction;

    public Ray(Vector3 origin, Vector3 direction)
    {
        Origin = origin;
        Direction = direction;
    }
}

class RayTracer
{
    private static IEnumerable<object> hittables;
    
    static RayTracer()
    {
        var list = new List<object>();
        list.Add(new object());
        hittables = list;
    }

    public static void Main(string[] args)
    {
        for (int x = 0; x < 100*100*100; x++)
        {
            Ray r = GetRay();
            System.Console.WriteLine(r.Direction);
            traceRay(r); 
        }
    }

    public static Ray GetRay()
    {
        return new Ray(Vector3.Zero, -Vector3.UnitY);
    }

    private static Vector3 traceRay(Ray r)
    {
        foreach (object h in hittables) {
            TestHit(r);
            return Vector3.One;
        }
        return Vector3.Zero;
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    public static void TestHit(Ray ray)
    {
        return;
    }
}
```

### Expected behavior

No memory corruption

### Actual behavior

Memory corruption, resulting in a `System.AccessViolationException`.

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

I'm running .NET 7 on macOS 10.15.7 (x64), code is being compiled in Release mode. I couldn't test it on other versions/platforms yet. Let me know if you need any further details of my setup.

```
% dotnet --version
7.0.100
% sw_vers
ProductName:    Mac OS X
ProductVersion: 10.15.7
BuildVersion:   19H2026
```

### Other information

There is a similar bug (#10972) that was fixed in .NET 3.

### LINKED ISSUE #88842: AccessViolationException or InvalidOperationException thrown in local method with .NET 8 preview 7 nightly
### Description

To check some fresh changes in the aspnetcore repo for Request Delegate Generator with interceptors, @captainsafia asked me to do some tests with the preview 7 nightlies.

In the course of that testing I updated a codebase that is working correctly .NET 7 and with `8.0.100-preview.6.23330.14` to version `8.0.100-preview.7.23363.2` of the SDK.

When pushing the changes to GitHub Actions CI, there's some weird failures happening only Linux. There's some code that's alternating between either failing with an `AccessViolationException` and crashing the test process, or throwing `InvalidOperationException` ~~and making the HTTP server _"go a bit weird"_~~.

The code for the method where the exceptions are coming from [can be found here](https://github.com/martincostello/adventofcode/blob/51f7c6398e2ce7a15192b71b3e16440f5b2f4edb/src/AdventOfCode/Puzzles/Y2021/Day22.cs#L32-L138) (don't judge it too harshly 😄).

### Reproduction Steps

1. Clone https://github.com/martincostello/adventofcode/pull/1094/commits/51f7c6398e2ce7a15192b71b3e16440f5b2f4edb on Linux
2. Run `build.ps1`

### Expected behavior

The application is successfully compiled and all the tests pass.

### Actual behavior

Either [an `AccessViolationException` is thrown](https://github.com/martincostello/adventofcode/actions/runs/5543769664/jobs/10120254091?pr=1094#step:10:877) and the test process crashes:

```
The active test run was aborted. Reason: Test host process crashed : Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at MartinCostello.AdventOfCode.Puzzles.Y2021.Day22.<Reboot>g__Reboot|8_1(System.Collections.Generic.List`1<System.ValueTuple`2<Cuboid,Boolean>>)
   at MartinCostello.AdventOfCode.Puzzles.Y2021.Day22.Reboot(System.Collections.Generic.IList`1<System.String>, Boolean)
   at MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+<SolveCoreAsync>d__9.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+<SolveCoreAsync>d__9, AdventOfCode, Version=8.0.0.0, Culture=neutral, PublicKeyToken=9a192a7522c9e1a0]](<SolveCoreAsync>d__9 ByRef)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].Start[[MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+<SolveCoreAsync>d__9, AdventOfCode, Version=8.0.0.0, Culture=neutral, PublicKeyToken=9a192a7522c9e1a0]](<SolveCoreAsync>d__9 ByRef)
   at MartinCostello.AdventOfCode.Puzzles.Y2021.Day22.SolveCoreAsync(System.String[], System.Threading.CancellationToken)
   at MartinCostello.AdventOfCode.Puzzle+<SolveAsync>d__18.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[MartinCostello.AdventOfCode.Puzzle+<SolveAsync>d__18, AdventOfCode, Version=8.0.0.0, Culture=neutral, PublicKeyToken=9a192a7522c9e1a0]](<SolveAsync>d__18 ByRef)
   at MartinCostello.AdventOfCode.Puzzles.PuzzleTest+<SolvePuzzleAsync>d__8`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[MartinCostello.AdventOfCode.Puzzles.PuzzleTest+<SolvePuzzleAsync>d__8`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], AdventOfCode.Tests, Version=8.0.0.0, Culture=neutral, PublicKeyToken=9a192a7522c9e1a0]](<SolvePuzzleAsync>d__8`1<System.__Canon> ByRef)
   at MartinCostello.AdventOfCode.Puzzles.PuzzleTest.SolvePuzzleAsync[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.String[])
   at MartinCostello.AdventOfCode.Puzzles.PuzzleTest+<SolvePuzzleAsync>d__7`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[MartinCostello.AdventOfCode.Puzzles.PuzzleTest+<SolvePuzzleAsync>d__7`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], AdventOfCode.Tests, Version=8.0.0.0, Culture=neutral, PublicKeyToken=9a192a7522c9e1a0]](<SolvePuzzleAsync>d__7`1<System.__Canon> ByRef)
   at MartinCostello.AdventOfCode.Puzzles.PuzzleTest.SolvePuzzleAsync[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]()
   at MartinCostello.AdventOfCode.Puzzles.Y2021.Day22Tests+<Y2021_Day22_Solve_Returns_Correct_Solution>d__4.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[MartinCostello.AdventOfCode.Puzzles.Y2021.Day22Tests+<Y2021_Day22_Solve_Returns_Correct_Solution>d__4, AdventOfCode.Tests, Version=8.0.0.0, Culture=neutral, PublicKeyToken=9a192a7522c9e1a0]](<Y2021_Day22_Solve_Returns_Correct_Solution>d__4 ByRef)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[MartinCostello.AdventOfCode.Puzzles.Y2021.Day22Tests+<Y2021_Day22_Solve_Returns_Correct_Solution>d__4, AdventOfCode.Tests, Version=8.0.0.0, Culture=neutral, PublicKeyToken=9a192a7522c9e1a0]](<Y2021_Day22_Solve_Returns_Correct_Solution>d__4 ByRef)
   at MartinCostello.AdventOfCode.Puzzles.Y2021.Day22Tests.Y2021_Day22_Solve_Returns_Correct_Solution()
   at System.RuntimeMethodHandle.InvokeMethod(System.Object, Void**, System.Signature, Boolean)
   at System.Reflection.MethodInvoker.Invoke(System.Object, IntPtr*, System.Reflection.BindingFlags)
   at System.Reflection.RuntimeMethodInfo.Invoke(System.Object, System.Reflection.BindingFlags, System.Reflection.Binder, System.Object[], System.Globalization.CultureInfo)
   at Xunit.Sdk.TestInvoker`1+<>c__DisplayClass48_0+<<InvokeTestMethodAsync>b__1>d[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.TestInvoker`1+<>c__DisplayClass48_0+<<InvokeTestMethodAsync>b__1>d[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], xunit.execution.dotnet, Version=2.5.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<<InvokeTestMethodAsync>b__1>d<System.__Canon> ByRef)
   at Xunit.Sdk.TestInvoker`1+<>c__DisplayClass48_0[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].<InvokeTestMethodAsync>b__1()
   at Xunit.Sdk.ExecutionTimer+<AggregateAsync>d__4.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.ExecutionTimer+<AggregateAsync>d__4, xunit.execution.dotnet, Version=2.5.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<AggregateAsync>d__4 ByRef)
   at Xunit.Sdk.ExceptionAggregator+<RunAsync>d__9.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.ExceptionAggregator+<RunAsync>d__9, xunit.core, Version=2.5.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<RunAsync>d__9 ByRef)
   at Xunit.Sdk.TestInvoker`1+<InvokeTestMethodAsync>d__48[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.TestInvoker`1+<InvokeTestMethodAsync>d__48[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], xunit.execution.dotnet, Version=2.5.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<InvokeTestMethodAsync>d__48<System.__Canon> ByRef)
   at Xunit.Sdk.TestInvoker`1+<<RunAsync>b__47_0>d[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.TestInvoker`1+<<RunAsync>b__47_0>d[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], xunit.execution.dotnet, Version=2.5.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<<RunAsync>b__47_0>d<System.__Canon> ByRef)
   at Xunit.Sdk.TestInvoker`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].<RunAsync>b__47_0()
   at Xunit.Sdk.ExceptionAggregator+<RunAsync>d__10`1[[System.Decimal, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.ExceptionAggregator+<RunAsync>d__10`1[[System.Decimal, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], xunit.core, Version=2.5.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<RunAsync>d__10`1<System.Decimal> ByRef)
   at Xunit.Sdk.XunitTestRunner+<InvokeTestAsync>d__4.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.XunitTestRunner+<InvokeTestAsync>d__4, xunit.execution.dotnet, Version=2.5.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<InvokeTestAsync>d__4 ByRef)
   at Xunit.Sdk.ExceptionAggregator+<RunAsync>d__10`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
```

Or [an `InvalidOperationException` is thrown from some tests](https://github.com/martincostello/adventofcode/actions/runs/5544169582/jobs/10121241278#step:10:882) ~~and then [the HTTP server stops responding](https://github.com/martincostello/adventofcode/actions/runs/5544169582/jobs/10121241278#step:10:1010)~~:

```
 System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
    at MartinCostello.AdventOfCode.Puzzles.Y2021.Day22.<Reboot>g__Reboot|8_1(List`1 cuboids) in /_/src/AdventOfCode/Puzzles/Y2021/Day22.cs:line 74
    at MartinCostello.AdventOfCode.Puzzles.Y2021.Day22.Reboot(IList`1 instructions, Boolean initialize) in /_/src/AdventOfCode/Puzzles/Y2021/Day22.cs:line 46
    at MartinCostello.AdventOfCode.Puzzles.Y2021.Day22.SolveCoreAsync(String[] args, CancellationToken cancellationToken) in /_/src/AdventOfCode/Puzzles/Y2021/Day22.cs:line 146
    at MartinCostello.AdventOfCode.Puzzle.SolveAsync(String[] args, CancellationToken cancellationToken) in /_/src/AdventOfCode/Puzzle.cs:line 57
    at MartinCostello.AdventOfCode.Site.PuzzlesApi.SolvePuzzleAsync(Int32 year, Int32 day, HttpRequest request, IFormFile resource, PuzzleFactory factory, ILogger`1 logger, CancellationToken cancellationToken) in /_/src/AdventOfCode.Site/PuzzlesApi.cs:line 117
```

### Regression?

Yes - the same code works in .NET 7 and .NET 8 preview 6.

### Known Workarounds

Set `DOTNET_JitEnablePhysicalPromotion=0`.

https://github.com/dotnet/runtime/issues/88842#issuecomment-1634520397

### Configuration

.NET SDK 8.0.100-preview.7.23363.2

### Other information

_No response_

_Edit_: Turns out the "weird server" bits aren't new, I just hadn't spotted them before.

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/dotnet/runtime/issues/79118
Fixes https://github.com/dotnet/runtime/issues/88842

### CODE REVIEW

**Comment by filipnavara** [comment]
> Fixes #88842 and, presumably, fixes #88842

Is this a joke on `a.SequenceEqual(a) == false` (aka issue #65292)?

Seriously though, I assumed you meant to link two different issues.

File: src/coreclr/jit/codegenlinear.cpp [hunk 1]
```diff
@@ -1868,13 +1868,18 @@ void CodeGen::genPutArgStkFieldList(GenTreePutArgStk* putArgStk, unsigned outArg
...
         {
-            // Need an additional integer register to extract upper 4 bytes from data.
+// Need an additional integer register to extract upper 4 bytes from data.
+#if defined(TARGET_ARM64)
+            // TODO: tmpReg can be avoided with st1 instruction on ARM64.
             regNumber tmpReg = nextArgNode->GetSingleTempReg();
             GetEmitter()->emitStoreSimd12ToLclOffset(outArgVarNum, thisFieldOffset, reg, tmpReg);
+#else
+            GetEmitter()->emitStoreSimd12ToLclOffset(outArgVarNum, thisFieldOffset, nextArgNode);
+#endif
```
Comment (jakobbotsch) [inline src/coreclr/jit/codegenlinear.cpp:1882]: Can we unify `emitStoreSimd12ToLclOffset` to have the same signature and only one declaration?

**Comment by jakobbotsch** [comment]
I verified that this fixes #88842 as well, diff:
```diff
@@ -674,27 +674,31 @@ G_M13415_IG07:
 						;; size=10 bbWeight=1595.95 PerfScore 1462.95
 G_M13415_IG08:
        test     edx, edx
-       je       SHORT G_M13415_IG11
-						;; size=4 bbWeight=99.75 PerfScore 124.68
+       je       G_M13415_IG11
+						;; size=8 bbWeight=99.75 PerfScore 124.68
 G_M13415_IG09:
        mov      gword ptr [rbp-78H], rax
        mov      dword ptr [rbp-70H], edi
        vmovaps  xmmword ptr [rbp+C0H], xmm4
        vmovups  xmmword ptr [rbp+CCH], xmm2
        cmp      byte  ptr [rbx], bl
-       vmovups  xmmword ptr [rsp], xmm4
-       vmovups  xmmword ptr [rsp+0CH], xmm2
+       vmovsd   qword ptr [rsp], xmm4
+       vextractps dword ptr [rsp+08H], xmm4, 2
+       vmovsd   qword ptr [rsp+0CH], xmm2
+       vextractps dword ptr [rsp+14H], xmm2, 2
        mov      rdi, rbx
        call     [System.Collections.Generic.Dictionary`2[MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid,long]:FindValue(MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid):byref:this]
        test     rax, rax
        je       SHORT G_M13415_IG12
-						;; size=50 bbWeight=36.60 PerfScore 494.10
+						;; size=66 bbWeight=36.60 PerfScore 713.70
 G_M13415_IG10:
        mov      rsi, qword ptr [rax]
        vmovaps  xmm4, xmmword ptr [rbp+C0H]
-       vmovups  xmmword ptr [rsp], xmm4
+       vmovsd   qword ptr [rsp], xmm4
+       vextractps dword ptr [rsp+08H], xmm4, 2
        vmovups  xmm2, xmmword ptr [rbp+CCH]
-       vmovups  xmmword ptr [rsp+0CH], xmm2
+       vmovsd   qword ptr [rsp+0CH], xmm2
+       vextractps dword ptr [rsp+14H], xmm2, 2
        sub      rsi, r13
        mov      rdi, rbx
        mov      edx, 1
@@ -705,7 +709,7 @@ G_M13415_IG10:
        vmovups  xmm0, xmmword ptr [rbp-58H]
        vmovups  xmm1, xmmword ptr [rbp-68H]
        jmp      SHORT G_M13415_IG13
-						;; size=69 bbWeight=585.60 PerfScore 14493.55
+						;; size=85 bbWeight=585.60 PerfScore 18007.13
 G_M13415_IG11:
        mov      ecx, dword ptr [rbp-6CH]
        vmovups  xmm0, xmmword ptr [rbp-58H]
@@ -714,9 +718,11 @@ G_M13415_IG11:
 						;; size=15 bbWeight=49.87 PerfScore 448.86
 G_M13415_IG12:
        vmovaps  xmm4, xmmword ptr [rbp+C0H]
-       vmovups  xmmword ptr [rsp], xmm4
+       vmovsd   qword ptr [rsp], xmm4
+       vextractps dword ptr [rsp+08H], xmm4, 2
        vmovups  xmm2, xmmword ptr [rbp+CCH]
-       vmovups  xmmword ptr [rsp+0CH], xmm2
+       vmovsd   qword ptr [rsp+0CH], xmm2
+       vextractps dword ptr [rsp+14H], xmm2, 2
        mov      rsi, r13
        neg      rsi
        mov      rdi, rbx
@@ -727,7 +733,7 @@ G_M13415_IG12:
        mov      edi, dword ptr [rbp-70H]
        vmovups  xmm0, xmmword ptr [rbp-58H]
        vmovups  xmm1, xmmword ptr [rbp-68H]
-						;; size=67 bbWeight=585.60 PerfScore 12297.56
+						;; size=83 bbWeight=585.60 PerfScore 15811.14
 G_M13415_IG13:
        mov      r13d, 1
        mov      dword ptr [rbp-6CH], ecx
@@ -769,13 +775,15 @@ G_M13415_IG18:
        vmovups  xmmword ptr [rbp-38H], xmm0
        vmovups  xmmword ptr [rbp-48H], xmm1
        cmp      byte  ptr [rbx], bl
-       vmovups  xmmword ptr [rsp], xmm0
-       vmovups  xmmword ptr [rsp+0CH], xmm1
+       vmovsd   qword ptr [rsp], xmm0
+       vextractps dword ptr [rsp+08H], xmm0, 2
+       vmovsd   qword ptr [rsp+0CH], xmm1
+       vextractps dword ptr [rsp+14H], xmm1, 2
        mov      rdi, rbx
        call     [System.Collections.Generic.Dictionary`2[MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid,long]:FindValue(MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid):byref:this]
        test     rax, rax
        je       G_M13415_IG25
-						;; size=41 bbWeight=0.19 PerfScore 2.15
+						;; size=57 bbWeight=0.19 PerfScore 3.28
 G_M13415_IG19:
        mov      rsi, qword ptr [rax]
        jmp      G_M13415_IG24
@@ -801,47 +809,55 @@ G_M13415_IG22:
        vmovups  xmmword ptr [rbp+A8H], xmm2
        vmovups  xmmword ptr [rbp+B4H], xmm3
        mov      qword ptr [rbp+A0H], rdx
-       vmovups  xmmword ptr [rsp], xmm2
-       vmovups  xmmword ptr [rsp+0CH], xmm3
+       vmovsd   qword ptr [rsp], xmm2
+       vextractps dword ptr [rsp+08H], xmm2, 2
+       vmovsd   qword ptr [rsp+0CH], xmm3
+       vextractps dword ptr [rsp+14H], xmm3, 2
        mov      rdi, r15
        call     [System.Collections.Generic.Dictionary`2[MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid,long]:FindValue(MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid):byref:this]
        test     rax, rax
        je       G_M13415_IG33
-						;; size=52 bbWeight=20.63 PerfScore 196.03
+						;; size=68 bbWeight=20.63 PerfScore 319.83
 G_M13415_IG23:
        mov      rsi, qword ptr [rax]
        vmovups  xmm2, xmmword ptr [rbp+A8H]
-       vmovups  xmmword ptr [rsp], xmm2
+       vmovsd   qword ptr [rsp], xmm2
+       vextractps dword ptr [rsp+08H], xmm2, 2
        vmovups  xmm3, xmmword ptr [rbp+B4H]
-       vmovups  xmmword ptr [rsp+0CH], xmm3
+       vmovsd   qword ptr [rsp+0CH], xmm3
+       vextractps dword ptr [rsp+14H], xmm3, 2
        add      rsi, qword ptr [rbp+A0H]
        mov      rdi, r15
        mov      edx, 1
        call     [System.Collections.Generic.Dictionary`2[MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid,long]:TryInsert(MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid,long,ubyte):bool:this]
        mov      eax, dword ptr [rbp-70H]
        jmp      G_M13415_IG34
-						;; size=59 bbWeight=330.15 PerfScore 6107.73
+						;; size=75 bbWeight=330.15 PerfScore 8088.61
 G_M13415_IG24:
        vmovups  xmm1, xmmword ptr [rbp-38H]
-       vmovups  xmmword ptr [rsp], xmm1
+       vmovsd   qword ptr [rsp], xmm1
+       vextractps dword ptr [rsp+08H], xmm1, 2
        vmovups  xmm2, xmmword ptr [rbp-48H]
-       vmovups  xmmword ptr [rsp+0CH], xmm2
+       vmovsd   qword ptr [rsp+0CH], xmm2
+       vextractps dword ptr [rsp+14H], xmm2, 2
        inc      rsi
        mov      rdi, rbx
        mov      edx, 1
        call     [System.Collections.Generic.Dictionary`2[MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid,long]:TryInsert(MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid,long,ubyte):bool:this]
        jmp      SHORT G_M13415_IG26
-						;; size=40 bbWeight=0.19 PerfScore 2.57
+						;; size=56 bbWeight=0.19 PerfScore 3.70
 G_M13415_IG25:
        vmovups  xmm1, xmmword ptr [rbp-38H]
-       vmovups  xmmword ptr [rsp], xmm1
+       vmovsd   qword ptr [rsp], xmm1
+       vextractps dword ptr [rsp+08H], xmm1, 2
        vmovups  xmm2, xmmword ptr [rbp-48H]
-       vmovups  xmmword ptr [rsp+0CH], xmm2
+       vmovsd   qword ptr [rsp+0CH], xmm2
+       vextractps dword ptr [rsp+14H], xmm2, 2
        mov      rdi, rbx
        mov      esi, 1
        mov      edx, 1
        call     [System.Collections.Generic.Dictionary`2[MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid,long]:TryInsert(MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid,long,ubyte):bool:this]
-						;; size=40 bbWeight=0.75 PerfScore 8.80
+						;; size=56 bbWeight=0.75 PerfScore 13.29
 G_M13415_IG26:
        mov      edi, dword ptr [rbx+38H]
        sub      edi, dword ptr [rbx+40H]
@@ -893,15 +909,17 @@ G_M13415_IG32:
 						;; size=22 bbWeight=0.23 PerfScore 1.33
 G_M13415_IG33:
        vmovups  xmm2, xmmword ptr [rbp+A8H]
-       vmovups  xmmword ptr [rsp], xmm2
+       vmovsd   qword ptr [rsp], xmm2
+       vextractps dword ptr [rsp+08H], xmm2, 2
        vmovups  xmm3, xmmword ptr [rbp+B4H]
-       vmovups  xmmword ptr [rsp+0CH], xmm3
+       vmovsd   qword ptr [rsp+0CH], xmm3
+       vextractps dword ptr [rsp+14H], xmm3, 2
        mov      rdi, r15
        mov      rsi, qword ptr [rbp+A0H]
        mov      edx, 1
        call     [System.Collections.Generic.Dictionary`2[MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid,long]:TryInsert(MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid,long,ubyte):bool:this]
        mov      eax, dword ptr [rbp-70H]
-						;; size=51 bbWeight=330.15 PerfScore 4456.99
+						;; size=67 bbWeight=330.15 PerfScore 6437.88
 G_M13415_IG34:
        cmp      r14d, dword ptr [rbx+44H]
        jne      G_M13415_IG47
@@ -1031,6 +1049,11 @@ G_M13415_IG48:
        int3     
 						;; size=6 bbWeight=0 PerfScore 0.00
 
-; Total bytes of code 1573, prolog size 84, PerfScore 71319.51, instruction count 350, allocated bytes for code 1573 (MethodHash=e316cb98) for method MartinCostello.AdventOfCode.Puzzles.Y2021.Day22:<Reboot>g__Reboot|8_1(System.Collections.Generic.List`1[System.ValueTuple`2[MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid,bool]]):long (Tier1-OSR)
+; Total bytes of code 1721, prolog size 84, PerfScore 82673.40, instruction count 368, allocated bytes for code 1721 (MethodHash=e316cb98) for method MartinCostello.AdventOfCode.Puzzles.Y2021.Day22:<Reboot>g__Reboot|8_1(System.Collections.Generic.List`1[System.ValueTuple`2[MartinCostello.AdventOfCode.Puzzles.Y2021.Day22+Cuboid,bool]]):long (Tier1-OSR)
```

File: src/coreclr/jit/emitxarch.h [hunk 2]
```diff
@@ -858,6 +858,10 @@ bool emitIsLastInsCall();
 void emitOutputPreEpilogNOP();
 #endif // TARGET_AMD64
 
+#if defined(FEATURE_SIMD)
+void emitStoreSimd12ToLclOffset(unsigned varNum, unsigned offset, regNumber dataReg, GenTree* tmpRegProvider);
+#endif // FEATURE_SIMD
```
Comment (jakobbotsch) [inline src/coreclr/jit/emitxarch.h:863]: Can we declare this in emit.h or somewhere common toi?

**Review by jakobbotsch — APPROVED** [review]

**Comment by EgorBo** [comment]
SPMI is failing everywhere, real diffs were +24 few cases, e.g.:
```asm
@@ -163,7 +163,9 @@ G_M13885_IG02:        ; bbWeight=1, gcrefRegs=380000 {x19 x20 x21}, byrefRegs=00
             fmul    v0.4s, v1.4s, v0.s[0]
             fsub    v0.4s, v16.4s, v0.4s
             str     q0, [fp, #0x20]	// [V50 tmp41]
-            str     q0, [sp]	// [V09 OutArgs]
+            str     d0, [sp]	// [V09 OutArgs]
+            mov     w1, v0.s[2]
+            str     w1, [sp, #0x08]	// [V09 OutArgs+0x08]
             ldp     s0, s1, [fp, #0xA0]	// [V24 tmp15], [V24 tmp15+0x04]
             ldr     s2, [fp, #0xA8]	// [V24 tmp15+0x08]
             ldp     s3, s4, [fp, #0xC0]	// [V06 loc2], [V06 loc2+0x04]
@@ -189,7 +191,7 @@ G_M13885_IG02:        ; bbWeight=1, gcrefRegs=380000 {x19 x20 x21}, byrefRegs=00
             cmp     w22, #5
             mov     v8.d[1], v9.d[0]
             blt     G_M13885_IG05
-						;; size=248 bbWeight=1 PerfScore 109.00
+						;; size=256 bbWeight=1 PerfScore 111.00
```
Because previously we only ran that logic on osx-arm64

**Pull request merged**